src/bdk_cli.rs

```diff
@@ -104,29 +116,40 @@ fn new_online_wallet<D>(
 where
     D: BatchDatabase,
 {
-    // Try to use Esplora config if "esplora" feature is enabled
-    #[cfg(feature = "esplora")]
-    let config_esplora: Option<AnyBlockchainConfig> = {
-        let esplora_concurrency = wallet_opts.esplora_opts.esplora_concurrency;
-        wallet_opts.esplora_opts.esplora.clone().map(|base_url| {
-            AnyBlockchainConfig::Esplora(EsploraBlockchainConfig {
-                base_url,
-                concurrency: Some(esplora_concurrency),
-            })
-        })
-    };
-    #[cfg(not(feature = "esplora"))]
-    let config_esplora = None;
-
-    let config_electrum = AnyBlockchainConfig::Electrum(ElectrumBlockchainConfig {
+    #[cfg(all(
+        feature = "electrum",
+        any(feature = "esplora", feature = "compact_filters")
+    ))]
+    compile_error!("Only one blockchain client feature can be enabled at a time.");
+
+    #[cfg(feature = "electrum")]
+    let config = AnyBlockchainConfig::Electrum(ElectrumBlockchainConfig {
         url: wallet_opts.electrum_opts.electrum.clone(),
         socks5: wallet_opts.proxy_opts.proxy.clone(),
         retry: wallet_opts.proxy_opts.retries,
         timeout: wallet_opts.electrum_opts.timeout,
     });
 
+    #[cfg(all(
+        feature = "esplora",
+        any(feature = "electrum", feature = "compact_filters")
+    ))]
+    compile_error!("Only one blockchain client feature can be enabled at a time.");
```

Comment:
It seems for every combination of `electrum`, `esplora` and `compact_filters`, this error is being duplicated with the one defined few lines below.

Instead, if we write it like this 
```
#[cfg(all(
        any( feature = "electrum", feature = "esplora", feature = "compact_filters"),
        any( feature = "electrum", feature = "esplora", feature = "compact_filters")
    ))]
```
this will activate for any combination of the above three. And we won't need to specify `compiler_error!` multiple times. 

Also for future extension, if we just add a new backend to the list above, that will taker care of multiple backend activation errors.

Reply:
Unfortunately the cfg all any any approach doesn't work, when I added this code:
```rust
#[cfg(all(
any( feature = "electrum", feature = "esplora", feature = "compact_filters"),
any( feature = "electrum", feature = "esplora", feature = "compact_filters")
))]
compile_error!("Only one blockchain client feature can be enabled at a time.");
```
I get the error even with only enable one blockchain client feature (ie. `cargo build --features esplora`). 

Looks like other's have had this same problem and are proposing a [Rust RFC](https://internals.rust-lang.org/t/pre-rfc-cargo-mutually-exclusive-features/13182/2) to allow a set of features to `provide` another feature in a mutually exclusive way.  

Until mutually exclusive features are possible I can fix this in a simplistic way in the `build.rs` file as below or just leave it as it.. What do you think?

```rust
fn main() {
    let electrum = env::var_os("CARGO_FEATURE_ELECTRUM").map(|_| "electrum".to_string());
    let esplora = env::var_os("CARGO_FEATURE_ESPLORA").map(|_| "esplora".to_string());
    let compact_filters = env::var_os("CARGO_FEATURE_COMPACT_FILTERS").map(|_| "compact_filters".to_string());

    let blockchain_features : Vec<String> = vec!(electrum, esplora, compact_filters)
        .iter()
        .map(|f| f.to_owned())
        .flatten()
        .collect();
    
    if blockchain_features.len() > 1 {
        panic!("At most one blockchain client feature can be enabled but these features were enabled: {:?}", blockchain_features)
    }
}
```

Reply:
Ah ok.. I thought it would work because my analyzer didn't throw. Yes there should be a way to handle mutually exclusive features in cargo itself.

I like the build script approach. It's clear and concise. Till it's available in cargo we can use a guard like above.